spanner/benchmarks/benchmarks.go

```diff
@@ -124,13 +126,17 @@ func executeQuery(ctx context.Context, client *spanner.Client) (int64, error) {
 		"p1": generateUniqueID(),
 	}})
 	for {
-		_, err := iter.Next()
+		row, err := iter.Next()
 		if errors.Is(err, iterator.Done) {
 			break
 		}
 		if err != nil {
 			return time.Duration(0).Microseconds(), err
 		}
+
+		if row.ColumnValue(0).GetStringValue() == "" {
+			return time.Duration(0).Microseconds(), errors.New("empty ID in the table")
+		}
```

Comment:
This does not decode the value, this just gets the protobuf string value. What you should do is this:

```
var id int64
if err := row.Columns(&id); err != nil {
	return err
}
```

That will check what type the column is, see that it is an `INT64`, and then convert the underlying string value to an int64.